sql-map-jquery-test/src/main/java/com/hazelcast/jet/tests/jquery/tests/AbstractJsonMapTest.java

```diff
@@ -0,0 +1,163 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.jquery.tests;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.jet.sql.impl.connector.map.IMapSqlConnector;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.jet.tests.common.Util;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_KEY_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_VALUE_FORMAT;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+public abstract class AbstractJsonMapTest extends AbstractSoakTest {
+
+    protected static final int DEFAULT_QUERY_TIMEOUT_MILLIS = 100;
+    private static final int PROGRESS_PRINT_QUERIES_INTERVAL = 500;
+    private static final String JSON_DATA_PATH_DEFAULT = "/home/ec2-user/ansible/dataFile.json";
+
+    protected final String mapName;
+    protected final String sqlQuery;
+    protected final Boolean resultRequiredSort;
+    protected final String expectedJsonResultString;
+    protected HazelcastInstance client;
+
+    protected int queryTimeout = propertyInt("queryTimeout", DEFAULT_QUERY_TIMEOUT_MILLIS);
+    protected String inputJsonFile = property("jsonDataFilePath", JSON_DATA_PATH_DEFAULT);
+
+    protected String jsonInputString;
+    private long begin;
+    private long currentQueryCount;
+    private long lastQueryCount;
+    private long lastProgressPrintCount;
+
+    public AbstractJsonMapTest(String mapName, String sqlQuery, String expectedJsonPath, Boolean resultRequiredSort)
+            throws IOException {
+        this.mapName = mapName;
+        this.sqlQuery = sqlQuery;
+        this.jsonInputString = readJsonFromFile(inputJsonFile);
+        this.expectedJsonResultString = retrieveExpectedJsonStructure(jsonInputString, expectedJsonPath);
+        this.resultRequiredSort = resultRequiredSort;
+    }
+
+    protected abstract String retrieveExpectedJsonStructure(String jsonInputString, String jsonPath);
+    protected void runTest() {
+        begin = System.currentTimeMillis();
+
+        SqlService sql = client.getSql();
+
+        String sqlString = "CREATE OR REPLACE MAPPING " + mapName +
+                " TYPE " + IMapSqlConnector.TYPE_NAME + "\n" +
+                "OPTIONS (\n" +
+                '\'' + OPTION_KEY_FORMAT + "'='bigint'\n" + "," +
```

Comment:
This part still seems strange: `"'='bigint'\n" + ","`. Moreover the result of it is for example like:
```
'keyFormat'='bigint'
,'valueFormat'='json'
```
It's not an issue but having comma at the start of line seem strange for me.

Reply:
Well in fact, the snippet here is from OS repo for SQL modules. Basically instead of providing my own solution, I reused this.
However since you did not like it I changed it as you wanted :)  . I used them as they seemed to be clear.
See tests: OS test_schemaEvolution_fieldRemoved.

Also createMapping method is used almost contains strings In form I used directly there and they
- look clear
- same style is Used in OS repo in many places
- contain colon at the beginneeng - another example here:

From OS repo:
public static void createMapping(HazelcastInstance instance, String name, Class<?> keyClass, Class<?> valueClass) {
        try (SqlResult result = instance.getSql().execute("CREATE OR REPLACE MAPPING " + name + " TYPE " + IMapSqlConnector.TYPE_NAME + "\n"
                + "OPTIONS (\n"
                + '\'' + OPTION_KEY_FORMAT + "'='" + JAVA_FORMAT + "'\n"
                + ", '" + OPTION_KEY_CLASS + "'='" + keyClass.getName() + "'\n"
                + ", '" + OPTION_VALUE_FORMAT + "'='" + JAVA_FORMAT + "'\n"
                + ", '" + OPTION_VALUE_CLASS + "'='" + valueClass.getName() + "'\n"
                + ")"
        )) {
            assertThat(result.updateCount()).isEqualTo(0);
        }
    }

Reply:
As I said, I don't know SQL part - it's mostly up to you - if it's common to have `,` and the start of line then feel free to use it in that way (this is just a minor thing).